### ISSUE TITLE
Minify HTML

### ISSUE THREAD

**Issue by blowsie** [ngbp/ngbp#341]
Is there any reason why the `index.html` is not minified, at least in the compiled version?

**Comment by joshdmiller** [comment]
Not really. It's one small file, so I hasn't bothered. But a pull request would be gladly accepted. :smile:

**Comment by julkue** [comment]
Well in fact compressing HTML is difficult because you are removing some whitespace you had in the development. Let's say you have two div-tags both with `display:inline-block` and both separated with a newline. Then you have a rendered whitespace between those two divs and you may optimize your layout for it. But in production you are removing this whitespace.
So in summary I think it's helpful but you should be careful when using inline-elements. However, I've implemented in another [boilerplate](https://github.com/julmot/jmBoilerplate).

**Comment by blowsie** [comment]
@julmot I'm confused, what issue are you trying to suggest here? "Compiled" code does not need to be human readable.

**Comment by julkue** [comment]
That's not what I'm talking about. I give you a short example:

``` html
<div style="display:inline-block;">Hello</div>
<div style="display:inline-block;">World</div>
```

This will have a blank between "Hello" and "World". However a HTML minification would generate:

``` html
<div style="display:inline-block;">Hello</div><div style="display:inline-block;">World</div>
```

This would remove the blank between those two words. You must be aware of this when using inline-block or inline-elements itself in layout issues combined with HTML minification. That's all.

Also when implementing a HTML minification into ngbp also `<script>`-elements with `type="text/ng-template"` need to be converted. If necessary you can look into the source code of jmBoilerplate to see how to achieve this: https://github.com/julmot/jmBoilerplate/blob/0.1.1/Gruntfile.js#L461
